File: 06679.cpp
#include <iostream>
using namespace std;

int sum_num(int n, int base) {
	int sum = 0;
	while (n) {
		sum += n % base;
		n /= base;
	}

	return sum;
}

int main() {
	cout << 2992 << endl;
	for (int i = 2993; i <= 9999; i++) {
		int ten = sum_num(i, 10);
		int twelve = sum_num(i, 12);
		int sixteen = sum_num(i, 16);

		if (ten == twelve && ten == sixteen)
			cout << i << endl;
	}
}

File: 06679.py
def sum_num(n, base):
    sum = 0
    while n:
        sum += n % base
        n //= base

    return sum

print(2992)
for i in range(2993, 10000):
    ten = sum_num(i, 10)
    twelve = sum_num(i, 12)
    sixteen = sum_num(i, 16)

    if ten == twelve and ten == sixteen:
        print(i)
